File: src/my_project_typescript/easy_problems_typescript/from1to50/islandPerimeter.ts
function islandPerimeter(grid: number[][]): number {

    let perimeter: number = 0;
// this is a new variable that I wont use
let variableOne: number = 1
    for (let i = 0; i < grid.length; i++){
        for (let j = 0; j < grid[0].length; j++){
            if (grid[i][j] ==1){
                perimeter += 4
                if (i > 0 && grid[i-1][j] == 1){
                    perimeter -= 2
                }
                if (j > 0 && grid[i][j-1]){
                    perimeter -= 2
                }
            }
        }
    }
    return perimeter

};

console.log(islandPerimeter([[0,1,0,0],[1,1,1,0],[0,1,0,0],[1,1,0,0]]))

File: src/my_java_projects/src/easy_problems/from_1_to_50/IslandPerimeter.java
package easy_problems.from_1_to_50;

public class IslandPerimeter {
    public int islandPerimeter(int[][] grid) {
        int r = grid.length;
        int c = grid[0].length;
        int count = 0;
        for(int i = 0; i < r; i++) {
            for (int j = 0; j < c; j++) {
                if (grid[i][j] == 1) {
                    count += (i + 1 >= r || grid[i + 1][j] == 0) ? 1 : 0;
                    count += (i - 1 < 0 || grid[i - 1][j] == 0) ? 1 : 0;
                    count += (j + 1 >= c || grid[i][j + 1] == 0) ? 1 : 0;
                    count += (j - 1 < 0 || grid[i][j - 1] == 0) ? 1 : 0;
                }
            }
        }
        return count;
    }
}
